src/minecraft/kotlin/xyz/wagyourtail/unimined/minecraft/patch/fabric/AccessWidenerMerger.kt

```diff
@@ -0,0 +1,452 @@
+        access: AccessWidenerReader.AccessType,
+        defaultAccess: Access
+    ) {
+        if (entry == null || access == null) {
+            throw RuntimeException("Input entry or access is null")
+        }
+        map[entry] = applyAccess(access, map.getOrDefault(entry, defaultAccess), entry)
+    }
+
+    fun applyAccess(
+        input: AccessWidenerReader.AccessType,
+        access: Access,
+        entryTriple: EntryTriple?
+    ): Access {
+        return when (input) {
+            AccessWidenerReader.AccessType.ACCESSIBLE -> {
+                makeClassAccessible(entryTriple)
+                access.makeAccessible()
+            }
+
+            AccessWidenerReader.AccessType.EXTENDABLE -> {
+                makeClassExtendable(entryTriple)
+                access.makeExtendable()
+            }
+
+            AccessWidenerReader.AccessType.MUTABLE -> access.makeMutable()
+            else -> throw java.lang.UnsupportedOperationException("Unknown access type:$input")
+        }
+    }
+
+    private fun makeClassAccessible(entryTriple: EntryTriple?) {
+        if (entryTriple == null) return
+        classAccess.put(
+            entryTriple.owner,
+            applyAccess(
+                AccessWidenerReader.AccessType.ACCESSIBLE,
+                classAccess.getOrDefault(entryTriple.owner, ClassAccess.DEFAULT),
+                null
+            )
+        )
+    }
+
+    private fun makeClassExtendable(entryTriple: EntryTriple?) {
+        if (entryTriple == null) return
+        classAccess.put(
+            entryTriple.owner,
+            applyAccess(
+                AccessWidenerReader.AccessType.EXTENDABLE,
+                classAccess.getOrDefault(entryTriple.owner, ClassAccess.DEFAULT),
+                null
+            )
+        )
+    }
+
+    fun getClassAccess(className: String): Access {
+        return classAccess.getOrDefault(className, ClassAccess.DEFAULT)
+    }
+
+    fun getFieldAccess(entryTriple: EntryTriple): Access {
+        return fieldAccess.getOrDefault(entryTriple, FieldAccess.DEFAULT)
+    }
+
+    fun getMethodAccess(entryTriple: EntryTriple): Access {
+        return methodAccess.getOrDefault(entryTriple, MethodAccess.DEFAULT)
+    }
+}
+
+class AccessWidenerBetterRemapper
+/**
+ * @param delegate      The visitor to forward the remapped information to.
+ * @param toNamespace   The namespace that the access widener will be remapped to.
+ */(
+    private val delegate: AccessWidenerVisitor,
+    private val mappingsProvider: MappingsProvider,
+    private val toNamespace: String
+) : AccessWidenerVisitor {
+    private var remapper: TinyRemapper
+
+    init {
+        remapper = TinyRemapper.newRemapper()
+            .withMappings(
+                mappingsProvider.getMappingsProvider(
+                    EnvType.COMBINED,
+                    MappingNamespace.OFFICIAL to MappingNamespace.getNamespace(toNamespace),
+                    false
+                )
+            ).build()
```

Comment:
the remapper needs to be primed with the mc jar in the source namespace so it can actually resolve inheritance for the mappings to be complete